content/docs/rfcs/17/README.md

```diff
@@ -182,6 +183,43 @@ Below is the description of the fields of `RateLimitProof` and their types.
 | `share_x` and `share_y`| array of 32 bytes each | Shamir secret shares of the user's secret identity key `sk` . `share_x` is the Poseidon hash of the `WakuMessage`'s `payload` concatenated with its `contentTopic` . `share_y` is calculated using [Shamir secret sharing scheme](/spec/32) | <!-- todo specify the poseidon hash setting -->
 | `nullifier`  | array of 32 bytes | internal nullifier derived from `epoch` and peer's `sk` as explained in [RLN construct](/spec/32)|
 
+## RLN Credentials format
+A RLN credential collects all the relevant fields that allow generation of valid RLN proofs with respect to an onchain membership contract.
+
+To make it compatible across different clients and platforms, RLN credentials will be encoded in JSON. 
+This further allows to validate credentials using [JSON schema](https://json-schema.org/).
+
+```
+{
+  "application": string,
+  "appIdentifier": string,
+  "credentials": [{
+    "key": string,
+    "commitment": string,
+    "membershipGroups" : [{
+      "chainId": number,
+      "contract": string,
+      "treeIndex": string
+    }]
+  }],
+  "version": number
+}
+```
+Fields are specified as follow:
+
+- `application` : the application name within which credential(s) refer to.
+- `appIdentifier` : a unique identifier for the application. In RLN-RELAY, this SHOULD corresond to the application-specific [RLN identifier](https://rfc.vac.dev/spec/32/#terminology).
```

Comment:
I am not sure whether we really need to persist the `application` and `application identifier` in this file. That can be provided by the application layer, as the membership info is not tied to that. 

To me, there are two important pieces of info to maintain: The RLN group info (G), and the membership info (M)within that RLN group. They have an n:n relationship, that is, a user can have multiple accounts (M) in a given group (G) hence n:1. Or the same account M in multiple groups Gs i.e., 1:n. Since the former is more likely i.e., n:1, I think it would be more efficient to have the group information as the key and an array of accounts as the value. WDYT?